Split diffs and discarding lines (#1324)
github/docs: content/desktop/contributing-and-collaborating-using-github-desktop/committing-and-reviewing-changes-to-your-project.md

{% note %}

**Note:** Split diff displays are currently in beta and subject to change.

{% endnote %}

1. In the top right corner, use {% octicon "gear" aria-label="The Gear icon" %} to select **Unified** or **Split** to review your changes.
  ![Gear icon with unified and split diffs](/assets/images/help/desktop/gear-diff-select.png)
2. The lines that will be included in the commit are highlighted in blue. To exclude the change, click one or more changed lines so the blue disappears.
  ![Unselected lines in a file](/assets/images/help/desktop/partial-commit.png)

### 3. Discarding changes
If you have uncommitted changes that you don't want, you can remove the changes by discarding them. You can discard all uncommitted changes in one file, a range of files, or discard all changes in all files since the last commit.

> Discarded changes are saved in a dated file in the Trash and you can recover them until the Trash is emptied.

Reviewer: ```suggestion
Discarded changes are saved in a dated file in the Trash. You can recover discarded changes until the Trash is emptied.
```

#### Discarding changes in one or more files

{% data reusables.desktop.select-discard-files %}
{% data reusables.desktop.click-discard-files %}
  ![Discard Changes option in context menu](/assets/images/help/desktop/discard-changes-mac.png)
{% data reusables.desktop.confirm-discard-files %}
  ![Discard Changes button in the confirmation dialog](/assets/images/help/desktop/discard-changes-confirm-mac.png)

> #### Discarding changes in one or more lines

Author: This was totally my fault 🙈 You can discard additions _or_ deletions, as long as you don't add and delete within the same "hunk" of code 🙃

> #### Discarding changes in one or more lines

Reviewer: Oohhhh that's handy!

> You can discard one or more added lines in a single file before you commit them.

Author: 🤔 I've got nothing better right now, so I'm going to roll with your suggestion 💖

{% note %}

> **Note:** Discarding single lines is disabled in a group of changes that adds and removes lines.

Author: I think this one should probably stay a note? When I was trying out the feature this caused me a lot of confusion.  If you have both additions and deletions in the same chunk of changes *discard changes* is greyed out, but I could not for the life of me figure out why. I think this could end up being a big driver of users coming to the docs.

> **Note:** Discarding single lines is disabled in a group of changes that adds and removes lines.

Reviewer: >Use callouts sparingly for high-value information - do not include general information, permissions, or prerequisites in callouts.

>Notes provide useful information or reminders for the user, but the information is not required to follow. Notes may not be relevant or necessary to every user.

IMO, this is both general information and necessary for every user to understand how the feature works. That being said, I'll trust you to make the call. 😊

{% endnote %}

In the list of changed files, right click on the line you want to discard and select **Discard added line**.

  ![Discard single line in the confirmation dialog](/assets/images/help/desktop/discard-single-line.png)

> In the list of changed files, right click the group of lines you want to discard and select **Discard added lines**. 

Author: @lecoursen It's a really small window of space that doesn't have a real definition. In this image, clicking on the section in the yellow square will give you an option to **Discard modified lines**. If you click anywhere left of the yellow square (in a change that doesn't add _and_ delete a line), you'd get **Discard modified line**. 
<img width="218" alt="Screen Shot 2020-11-16 at 2 22 18 PM" src="https://user-images.githubusercontent.com/9817819/99310050-a6a32280-2817-11eb-9536-4b8ebb56be47.png">

@billygriffin @ampinsk Any thoughts on how to clarify this to users?

> In the list of changed files, right click the group of lines you want to discard and select **Discard added lines**. 

Reviewer 2: What about just including a screenshot like you have above with the yellow square @janiceilene? I know it's suboptimal but that would make it much clearer to show people that they need to click in the thin part to select the entire group. Just a thought.

> In the list of changed files, right click the group of lines you want to discard and select **Discard added lines**. 

Reviewer: Oh wow... okay yes, that works now! Maybe something like this? I also love the idea of masking the bar in the screenshot, too!

```suggestion
To discard a group of changed lines, right click the vertical bar to the right of the line numbers for the lines you want to discard, then select **Discard added lines**.
```

>   ![Discard multiple lines in the confirmation dialog](/assets/images/help/desktop/discard-multiple-lines.png)

Reviewer: I think it's misleading to call it "multiple added lines" because that could imply removing lines that are far apart, like L5 and L100, (as you can with files)... when really they have to be grouped together.

```suggestion
  ![Discard a group of added lines in the confirmation dialog](/assets/images/help/desktop/discard-multiple-lines.png)
```
